c7n/resources/ecs.py

```diff
@@ -247,6 +247,90 @@ class ServiceTaskDefinitionFilter(RelatedTaskDefinitionFilter):
     """
 
 
+@Service.action_registry.register('modify')
+class UpdateService(BaseAction):
+    """Action to update service
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: no-public-ips-services
+                resource: ecs-service
+                filters:
+                  - 'networkConfiguration.awsvpcConfiguration.assignPublicIp': 'ENABLED'
+                actions:
+                  - type: modify
+                    update:
+                      networkConfiguration:
+                        awsvpcConfiguration:
+                          assignPublicIp: DISABLED
+    """
+
+    schema = type_schema('modify',
+        update={
```

Comment:
also, per @kapilt comment, why are you not exposing the actual parameters as top level keys? imo it's a lot cleaner to do:

```yaml
actions:
  - type: modify
    networkConfiguration:
      awsvpcConfiguration:
        assignPublicIp: DISABLED
```

you can still iterate over self.data.items() and remove type before submitting the api call

Reply:
tbh. i'm actually fine with this under the sub key/update. we have few others doing that as it allows the action to be extended with parameters that are not related to the resource attribute update, but to the actions operation. its also a little simpler with shape validation to have it self-contained, yeah you can manually disambiguate on all of these but its pretty nominal to just wrap it under key. per zen of py, namespaces are one honking good idea ;-)